packages/synapse-react-client/src/components/QueryWrapper/useQueryWrapperData.ts

```diff
@@ -0,0 +1,167 @@
+import React from 'react'
+import {
+  useGetQueryResultBundleWithAsyncStatus,
+  useInfiniteQueryResultBundle,
+} from '../../synapse-queries'
+import {
+  AsynchronousJobStatus,
+  QueryBundleRequest,
+  QueryResultBundle,
+} from '@sage-bionetworks/synapse-types'
+import { useMemo, useState } from 'react'
+import { SynapseClientError } from '../../utils/SynapseClientError'
+import { UseInfiniteQueryResult } from 'react-query'
+
+type InfiniteDataPage = number | 'ALL'
+
+type InfiniteDataQueryPageControls = Pick<
+  UseInfiniteQueryResult<
+    AsynchronousJobStatus<QueryBundleRequest, QueryResultBundle>,
+    SynapseClientError
+  >,
+  | 'fetchNextPage'
+  | 'fetchPreviousPage'
+  | 'hasNextPage'
+  | 'hasPreviousPage'
+  | 'isFetchingNextPage'
+> & {
+  currentPage: InfiniteDataPage
+  setCurrentPage: React.Dispatch<React.SetStateAction<InfiniteDataPage>>
+}
+
+export type UseQueryWrapperDataReturn<TIsInfiniteData extends boolean> = {
+  data?: QueryResultBundle
+  currentAsyncStatus?: AsynchronousJobStatus<
+    QueryBundleRequest,
+    QueryResultBundle
+  >
+  error: SynapseClientError | null
+  isLoadingNewBundle: boolean
+} & (TIsInfiniteData extends true ? InfiniteDataQueryPageControls : object)
```

Comment:
Taking advantage of conditional types on generics; if we have infinite data, the return object will also include the `useInfiniteQuery` pagination controls.